File: PriceChecker.Core/Models/ProductPrice.cs
using System.Text.Json.Serialization;

namespace Genius.PriceChecker.Core.Models;

public class ProductPrice
{
    public Guid ProductSourceId { get; set; }
    [JsonIgnore]
    public ProductSource ProductSource { get; set; } = null!; // Is being initialized in `repo.FillUpRelations`
    public AgentHandlingStatus Status { get; set; }
    public decimal? Price { get; set; }
    public DateTime FoundDate { get; set; }

    public override string ToString()
    {
        return $"{ProductSource?.Product?.Name ?? ProductSourceId.ToString()} = {Price}";
    }
}

File: PriceChecker.Web/src/models/productPrice.ts
import { AgentHandlingStatus } from "./enums";
import { ProductPriceRef, ProductSourceRef } from "./types";

interface ProductPrice {
    id: ProductPriceRef;
    productSourceId: ProductSourceRef;
    status: AgentHandlingStatus;
    price: number | null;
    foundDate: number;
}

export default ProductPrice;
